scripts/notifications/helpers.py

```diff
@@ -0,0 +1,251 @@
+            return et
+
+    return None
+
+
+def create_endpoint(name, xrhid, properties, ep_type="webhook", ep_subtype= None):
+    """Creates an endpoint"""
+
+    ep_uuid = uuid.uuid4()
+    ep_id = str(ep_uuid)
+    properties["endpointId"] = ep_id
+    ep_json = {"name": name,
+               "description": name,
+               "enabled": True,
+               "properties": properties,
+               "type": ep_type}
+    if ep_subtype is not None:
+        ep_json["sub_type"] = ep_subtype
+
+    h = {"x-rh-identity": xrhid}
+
+    r = requests.post(integrations_prefix + "/endpoints", json=ep_json, headers=h)
+    print(r.status_code)
+    if r.status_code / 100 != 2:
+        print(r.reason)
+        exit(1)
+
+    response_json = r.json()
+    epid = response_json["id"]
+    print(epid)
+
+    return epid
+
+
+def find_behavior_group(display_name, bundle_id, x_rhid):
+    """Find a behavior group by its display name"""
+
+    headers = {"x-rh-identity": x_rhid}
+    r = requests.get(notifications_prefix + "/notifications/bundles/" + bundle_id + "/behaviorGroups",
+                     headers=headers)
+
+    if r.status_code != 200:
+        return None
+
+    j = r.json()
+    for bg in j:
+        if bg["display_name"] == display_name:
+            return bg["id"]
+
+    return None
+
+
+def create_behavior_group(name, bundle_id, x_rhid):
+    """Creates a behavior group"""
+
+    bg_id = find_behavior_group(name, bundle_id, x_rhid)
+    if bg_id is not None:
+        return bg_id
+
+    bg_json = {"display_name": name,
+               "bundle_id": bundle_id}
+
+    headers = {"x-rh-identity": x_rhid}
+    r = requests.post(notifications_prefix + "/notifications/behaviorGroups",
+                      json=bg_json,
+                      headers=headers)
+
+    print(r.status_code)
+    if r.status_code / 100 != 2:
+        print(r.reason)
+        exit(1)
+
+    response_json = r.json()
+    bg_id = response_json["id"]
+    print(bg_id)
+
+    return bg_id
+
+
+def link_bg_endpoint(bg_id, ep_id, x_rhid):
+    """Link the behavior group to the endpoint"""
+    headers = {"x-rh-identity": x_rhid}
+
+    ep_list = [ep_id]
+
+    r = requests.put(notifications_prefix + "/notifications/behaviorGroups/" + bg_id + "/actions",
+                     json=ep_list,
+                     headers=headers)
+
+
+def add_endpoint_to_event_type(event_type_id, endpoint_id, x_rhid):
+    headers = {"x-rh-identity": x_rhid}
+    r = requests.put(notifications_prefix + "/notifications/eventTypes/" + event_type_id + "/" + endpoint_id,
+                     headers=headers)
+
+    print(r.status_code)
+
+def add_event_type_to_behavior_group(et_id, bg_id, x_rh_id):
+    bg_set = [bg_id]
+
+    headers = {"x-rh-identity": x_rh_id}
+    r = requests.put(notifications_prefix + "/notifications/eventTypes/" + et_id + "/behaviorGroups",
+                     json=bg_set,
+                     headers=headers)
+
+    print(r.status_code)
+
+    return None
```

Comment:
Is this your code or taken from somewhere?

Reply:
this is taken from the `notification-backend` repo, but it's not needed anymore